src/Data/Integer/Properties.agda

```diff
@@ -1400,26 +1390,10 @@ private
   rewrite +-identityʳ y
         | +-identityʳ (sign y Sign.* sign x ◃ ∣ y ∣ ℕ.* ∣ x ∣)
         = refl
-*-distribʳ-+ -[1+ m ] -[1+ n ] -[1+ o ] = cong (+_) $
-  solve 3 (λ m n o → (con 2 :+ n :+ o) :* (con 1 :+ m)
-                  := (con 1 :+ n) :* (con 1 :+ m) :+
-                     (con 1 :+ o) :* (con 1 :+ m))
-          refl m n o
-*-distribʳ-+ +[1+ m ] +[1+ n ] +[1+ o ] = cong (+_) $
-  solve 3 (λ m n o → (con 1 :+ n :+ (con 1 :+ o)) :* (con 1 :+ m)
-                  := (con 1 :+ n) :* (con 1 :+ m) :+
-                     (con 1 :+ o) :* (con 1 :+ m))
-        refl m n o
-*-distribʳ-+ -[1+ m ] +[1+ n ] +[1+ o ] = cong -[1+_] $
-  solve 3 (λ m n o → m :+ (n :+ (con 1 :+ o)) :* (con 1 :+ m)
-                   := (con 1 :+ n) :* (con 1 :+ m) :+
-                      (m :+ o :* (con 1 :+ m)))
-         refl m n o
-*-distribʳ-+ +[1+ m ] -[1+ n ] -[1+ o ] = cong -[1+_] $
-  solve 3 (λ m n o → m :+ (con 1 :+ m :+ (n :+ o) :* (con 1 :+ m))
-                  := (con 1 :+ n) :* (con 1 :+ m) :+
-                     (m :+ o :* (con 1 :+ m)))
-         refl m n o
+*-distribʳ-+ -[1+ m ] -[1+ n ] -[1+ o ] = cong (+_) $ assoc₁ m n o
```

Comment:
I'm not sure whether having half the cases in one file and the other set of equally long cases in this file, is ideal? I think it would be better (everywhere!) just to expand out the cases locally?

I know GCD.Lemmas doesn't do so, but that is a very old file?

Reply:
I'm fine with pulling these in.

I do think if we had a big `Lemmas` file, we'd notice quite a few patterns, and could shrink things. But that can be a v3.0 kind of task.

Reply:
And I now remember why I put these in a separate file: inside `Integer.Properties`, `+` and `*` are for integers, and the ones for naturals are 'long'. So these proofs end up looking much, much worse. But inside a different context, i.e. such as a separate module, this is not a problem.

So, in looking again, I'd rather not do this.

Reply:
Okay, I withdraw my objections.